.github/workflows/release.yaml

```
@@ -0,0 +1,63 @@
+name: Build OpenSearch from source, run tests and release.
+
+env:
+  RELEASE: edge
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  ci-tests:
+    name: Build and Run Tests
+    uses: ./.github/workflows/ci.yaml
+
+  publish:
+    name: publish
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
+    needs:
+      - ci-tests
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
+
+      - name: Install required dependencies
+        run: |
+          # docker
+          sudo snap install docker
+          sudo addgroup --system docker; sudo adduser $USER docker
+          newgrp docker
+          sudo snap disable docker; sudo snap enable docker
+
+          # skopeo
+          sudo snap install --devmode --channel edge skopeo
+          
+          # yq
+          sudo snap install yq
+
+      - name: Download rock file
+        uses: actions/download-artifact@v3
+        with:
+          name: charmed_opensearch_rock_amd64
+          path: .
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Publish rock to Store
+        run: |
+          version="$(cat rockcraft.yaml | yq .version)"
+          
+          sudo skopeo \
+              --insecure-policy \
+              copy \
+              oci-archive:charmed-opensearch_${version}_amd64.rock \
+              docker-daemon:charmed-opensearch:${version}
```

```suggestion
              docker-daemon:ghcr.io/charmed-opensearch:${version}
```
Played around with this on another repo.  I suffered so you dont have to.

---

You'll likely run into this [issue](https://stackoverflow.com/questions/69014742/github-denied-permission-denied-write-package) as well.  Someone with admin will need to change the permissions. @taurus-forever ?

---

GHCR_TOKEN secret added here and to mongodb-rock, please try it from your side.

---

@taurus-forever I think using the `GITHUB_TOKEN` with `admin` permissions may suffice, [for reference](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry#authenticating-in-a-github-actions-workflow)
> You can use a GITHUB_TOKEN in a GitHub Actions workflow to delete or restore a package using the REST API, if the token has admin permission to the package. Repositories that publish packages using a workflow, and repositories that you have explicitly connected to packages, are automatically granted admin permission to packages in the repository.